examples/tests/sdk_tests/website/aws/website_domain.w

```diff
@@ -0,0 +1,30 @@
+/*\
+skip: true
+\*/
+// this example only works on AWS (intentionally)
+
+bring cloud;
+bring http;
+bring aws;
+
+let domain = "www.example.com";
+
+/* The values of domain, hostedZoneId, and acmCertificateArn are fictitious
+ * only for the purpose of illustrating how to configure a domain for a 
+ * website on AWS */
+let w = new cloud.Website(aws.AwsWebsiteProps {
+  path: "./website",
+  domain: domain,
+  hostedZoneId: "Z111111111AAAA1A111AA",
+  acmCertificateArn: "arn:aws:acm:us-east-1:111111111111:certificate/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+});
```

Comment:
Some thoughts/questions:

1. Will it be confusing to users that they're passing a different props object than the one belonging to the class they're instantiating? 
2.  If I understand correctly, passing these props will still be valid if you're compiling to the simulator (they'll just get ignored). This is powerful, though it might not be obvious to the user which fields will get ignored or not. If we go with this approach, maybe we ought to have a convention where each target-specific field is prefixed with the target name, e.g. `awsHostedZoneId`.

We still need to figure out what are the right set of patterns for augmenting Wing Cloud Library classes with "target-specific" behavior, so it's great that we can use `cloud.Website` to try out these ideas.

@marciocadev Here's one thought I had - what if we create some kind of new preflight class that is only available through the `bring aws` that represents a `Domain`, and it can be instantiated with a `cloud.Website` and a list of AWS-specific props?

```js
bring cloud;
bring aws;

let w = new cloud.Website(path: "./website", domain: "boldbytes.co");
let d = new aws.Domain(w,
  hostedZoneId: "Z111111111AAAA1A111AA",
  acmCertificateArn: "arn:aws:acm:us-east-1:111111111111:certificate/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
);
```

WDYT of this API?

Pinging @eladb @hasanaburayyan since this could be related to platform libraries